packages/core/rpc/BaseRpcDriver.ts

```diff
@@ -188,10 +193,33 @@ export default abstract class BaseRpcDriver {
       pluginManager,
       sessionId,
     )
-    const result = await worker.call(functionName, filteredAndSerializedArgs, {
+
+    // now actually call the worker
+    const callP = worker.call(functionName, filteredAndSerializedArgs, {
       timeout: 5 * 60 * 1000, // 5 minutes
       ...options,
     })
-    return rpcMethod.deserializeReturn(result)
+
+    // check every 5 seconds to see if the worker has been killed, and
+    // reject the killedP promise if it has
+    let killedCheckInterval: ReturnType<typeof setInterval>
+    const killedP = new Promise((resolve, reject) => {
+      killedCheckInterval = setInterval(() => {
+        // must've been killed
+        if (worker.status === 'killed') {
+          reject(new Error('operation timed out'))
```

Comment:
maybe change the message to something a little more descriptive like "operation timed out, worker process stopped responding"